File: Codigos_em_Go/Euler_Serial.go
package main

import ("fmt")


func Fatorial(n float64) float64{
  if n == 0 {
    return 1;
  }else{
    return n*Fatorial(n-1)
  }
}

func Calc(){
  var e float64
  e = 0
  var i float64
  for i =0; i<10000; i++{
      e = e + 1/Fatorial(i)
  }
  fmt.Printf("%.30f\n",e)
}

func main() {
	Calc()
}

File: Codigos_em_C/Euler_Serial.c
#include<stdlib.h>
#include<stdio.h>

double Fatorial(int x);
void Calc();

int main(){
        Calc();
        return 0;
}


void Calc(){
        double e=0;
        for(int i=0; i<100000;i++){
                e = e + 1/(Fatorial(i));
        }

        printf("%.15f\n",e);

}

double Fatorial(int x){
        if(x == 0) return 1;
        else
                return x*Fatorial(x-1);

}
